server/server.ts

```diff
@@ -31,6 +34,34 @@ app.get('/', (_req: Request, res: Response) => {
     .sendFile(path.resolve(import.meta.dirname, '../client/index.html'));
 });
 
+app.post(
+  '/api',
+  // userQueryController,
+  openaiNLPQuery,
+  googlePlacesApi,
+  openAIRecommendationResponse,
+  //logger?
```

Comment:
@nanyi-deng This is where we should be including that `insertUserDataMiddleware` middleware function, so it has access to the data on `res.locals`